### PR TITLE
Update ESLint config to v7.0.1

### PR DESCRIPTION
**Pull request by Gudahtt**
The ESLint config has been updated, which includes the addition of `prettier. All lint issues have been fixed with `yarn lint:fix`. The ignore file was replaced by the ignore patterns property as well.

The Prettier plugin `prettier-plugin-packagejson` has been added to sort `package.json`.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -27,5 +20,5 @@ module.exports = {
     ],
     'no-param-reassign': 'off',
   },
-  ignorePatterns: ['!.eslintrc.js', './test/*.js', 'dist'],
+  ignorePatterns: ['!.eslintrc.js', 'test/*.js', 'dist'],
```
Comment (shanejonas) [inline .eslintrc.js:23]: should we add back the `coverage` and `*.d.ts` from the removed `.eslintignore` . And the `node_modules` ignore should maybe go in `tsconfig`?

Comment (Gudahtt) [reply .eslintrc.js:23]: Ah, good catch! But it looks like all of these are fine to ignore. `node_modules` is [ignored by default](https://eslint.org/docs/user-guide/configuring/ignoring-code#the-eslintignore-file), and we don't have any `coverage` directory or `*.d.ts` files in this repo.

**Comment by Gudahtt** [comment]
Just rebased this to fix lint errors introduced when the branch was updated.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**